apps/language_server/lib/language_server/experimental/provider/code_action/replace_remote_function.ex

```diff
@@ -0,0 +1,147 @@
+  alias ElixirLS.LanguageServer.Experimental.CodeMod.Ast
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.CodeAction, as: CodeActionResult
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.Diagnostic
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.TextEdit
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.Workspace
+  alias ElixirLS.LanguageServer.Experimental.SourceFile
+
+  @pattern ~r/(.*)\/(.*) is undefined or private. .*:\n(.*)/s
+
+  @spec pattern() :: Regex.t()
+  def pattern, do: @pattern
+
+  @spec apply(SourceFile.t(), Diagnostic.t()) :: [CodeActionResult.t()]
+  def apply(source_file, diagnostic) do
+    with {:ok, module, name} <- extract_function(diagnostic.message),
+         {:ok, suggestions} <- extract_suggestions(diagnostic.message),
+         one_based_line = extract_line(diagnostic),
+         {:ok, replies} <-
+           build_code_actions(source_file, one_based_line, module, name, suggestions) do
+      replies
+    else
+      _ ->
+        []
+    end
+  end
+
+  defp extract_function(message) do
+    case Regex.scan(@pattern, message) do
+      [[_, full_name, _, _]] ->
+        {module, name} = separate_module_from_name(full_name)
+        {:ok, module, name}
+
+      _ ->
+        :error
+    end
+  end
+
+  defp separate_module_from_name(full_name) do
+    {name, module} =
+      full_name
+      |> String.split(".")
+      |> Enum.map(&String.to_atom/1)
+      |> List.pop_at(-1)
+
+    {module, name}
+  end
+
+  @suggestion_pattern ~r/\* .*\/[\d]+/
+  defp extract_suggestions(message) do
+    case Regex.scan(@pattern, message) do
+      [[_, _, arity, suggestions_string]] ->
+        suggestions =
+          @suggestion_pattern
+          |> Regex.scan(suggestions_string)
+          |> Enum.flat_map(fn [suggestion] ->
+            case String.split(suggestion, [" ", "/"]) do
+              ["*", name, ^arity] -> [String.to_atom(name)]
+              _ -> []
+            end
+          end)
+
+        {:ok, suggestions}
+
+      _ ->
+        :error
+    end
+  end
+
+  defp extract_line(%Diagnostic{} = diagnostic) do
+    diagnostic.range.start.line
+  end
+
+  defp build_code_actions(%SourceFile{} = source_file, one_based_line, module, name, suggestions) do
+    with {:ok, line_text} <- SourceFile.fetch_text_at(source_file, one_based_line),
+         {:ok, line_ast} <- Ast.from(line_text),
+         {:ok, edits_per_suggestion} <-
+           text_edits_per_suggestion(line_text, line_ast, module, name, suggestions) do
+      case edits_per_suggestion do
+        [] ->
+          :error
+
+        [_ | _] ->
+          edits_per_suggestion =
+            Enum.map(edits_per_suggestion, fn {text_edits, suggestion} ->
+              text_edits = Enum.map(text_edits, &update_line(&1, one_based_line))
+              {text_edits, suggestion}
+            end)
+
+          replies =
+            Enum.map(edits_per_suggestion, fn {text_edits, function_name} ->
```

Comment:
the variable is called `edits_per_suggestion` but in the inner function, it's called `function_name`

Reply:
I will rename `function_name` with `suggestion`